nexus/src/authn/mod.rs

```diff
@@ -55,9 +55,18 @@ pub struct Context {
 impl Context {
     /// Returns the authenticated actor, if any
     pub fn actor(&self) -> Option<&Actor> {
+        self.actor_required().ok()
+    }
+
+    /// Returns the authenticated actor if present, Unauthenticated error otherwise
+    pub fn actor_required(&self) -> Result<&Actor, dropshot::HttpError> {
         match &self.kind {
-            Kind::Unauthenticated => None,
-            Kind::Authenticated(Details { actor }) => Some(actor),
+            Kind::Authenticated(Details { actor }) => Ok(actor),
+            Kind::Unauthenticated => Err(dropshot::HttpError::from(
+                omicron_common::api::external::Error::Unauthenticated {
+                    internal_message: "Actor required".to_string(),
+                },
+            )),
```

Comment:
I tried extracting this into a local function with this signature, but it didn't add much.

```rs
fn unauthenticated(internal_message: &str) -> dropshot::HttpError
```

Plus it felt like if I'm doing that, it might as well be over in common like the others:

https://github.com/oxidecomputer/omicron/blob/083ec19ea28731426cf5f214bfb3648550fbcb3a/common/src/api/external/error.rs#L151-L153